client/src/store/reducer/employeeTableReducer.js

```diff
@@ -0,0 +1,169 @@
+const initalState = {
+  employeeData: undefined,
+  filteredEmployeeData: undefined,
+  filters: {
+    department: {},
+    office: {},
+    name: '',
+  },
+  recipients: [],
+  page: 0,
+  rowsPerPage: 10,
+};
+
+const toggleSingleRecipient = (array, item) => {
+  const itemIndex = array.indexOf(item);
+  if (itemIndex === -1) {
+    return [...array, item];
+  }
+
+  return array.filter((x) => x !== item);
+};
+const toggleRecipients = (array, checked, recipientsList) => {
+  let updatedArray = [...array];
+  if (checked) {
+    recipientsList.forEach((item) => {
+      const itemIndex = array.indexOf(item);
+      if (itemIndex === -1) {
+        updatedArray = [...updatedArray, item];
+      }
+    });
+  } else {
+    recipientsList.forEach((item) => {
+      updatedArray = updatedArray.filter((x) => x !== item);
+    });
+  }
+  return updatedArray;
+};
+
+const setInitialOptions = (employeeData, attribute) => {
+  const onlyUnique = (value, index, self) => {
+    return self.indexOf(value) === index;
+  };
+
+  const allAttributeOptions = employeeData.map((person) => person[attribute]);
+
+  const uniqueAttributeOptions = allAttributeOptions.filter(onlyUnique);
+  const attributeCheckboxOptions = {};
+  uniqueAttributeOptions.forEach((option) => {
+    attributeCheckboxOptions[option] = false;
+  });
+  return attributeCheckboxOptions;
+};
+
+const setOption = (currentFilters, checked, option, attribute) => {
+  const updatedFilters = { ...currentFilters };
+  updatedFilters[attribute][option] = checked;
+  return updatedFilters;
+};
+
+const setAllOptions = (attributeCheckboxOptions, checked) => {
+  const newOptionsArray = { ...attributeCheckboxOptions };
```

Comment:
This isn't an array! And we should try to avoid referring to a variable's type in its name